### PR TITLE
Chore // added ability to be used on private channels

### PR DESCRIPTION
**Pull request by stuartjohnpage**
### Overview

Once we merge this, we'll have to reinstall the app to the workspace 👍 

Currently, Slax can't be used on private channels. This adds that functionality

- Adds `types=public_channel,private_channel` so we can retrieve these channels and information about them (ids, etc)
- Replaces `channels_select` (public only) with `multi_conversations_select` and filters so we still only get channels, not other conversation types
- Adds two permissions: 
  - groups:read - can access channel information (for multi_conversations_select)
  -  groups:history - so Slax will trigger when something like `example_repo_4_slax#1` is used

### CODE REVIEW

File: lib/slax_web/websockets/default_repo.ex [hunk 1]
```diff
@@ -68,13 +68,18 @@ defmodule SlaxWeb.DefaultRepo do
...
             placeholder: %{
               type: "plain_text",
               text: "Select channel",
               emoji: true
             },
-            action_id: "channels_select_action"
+            action_id: "channels_select_action",
+            filter: %{
+              include: ["public", "private"],
+              exclude_external_shared_channels: true,
```
Comment (Dacello) [inline lib/slax_web/websockets/default_repo.ex:80]: so we could set this to true and itd work for vetavize?

Comment (stuartjohnpage) [reply lib/slax_web/websockets/default_repo.ex:80]: perhaps - this is what populates the channel dropdown when we're setting default repos for channels. Maybe I should set it to false; because there's no guarantee all the other stuff would work for an externally shared channel. 

IIRC, we'd have to add Slax to that workspace, and then there might be some shenanigans involving the oauth permissions for external channels.

TL;DR: ...maybe? 😹

Comment (stuartjohnpage) [reply lib/slax_web/websockets/default_repo.ex:80]: I'm changing it to false

Comment (Dacello) [reply lib/slax_web/websockets/default_repo.ex:80]: word nbd ty for explain

**Review by Dacello — APPROVED** [review]

**Comment by dweill** [comment]
I'm not able to get slax to attempt to grab an issue from my private channel.
Screenshot 2024-08-08 at 11 42 33 AM

**Comment by dweill** [comment]
Screenshot 2024-08-08 at 11 43 20 AM
private channels appear in the dropdown, but I don't know what I need to do differently to get it to respond in private channels

**Comment by dweill** [comment]
Screenshot 2024-08-08 at 12 34 47 PM

@stuartjohnpage and I ended up needing to add this additional event subscription to get it to work in private channels.

**Review by dweill — APPROVED** [review]

**Comment by stuartjohnpage** [comment]
I think I should close this for now.

The reason I opened this was because there was a private channel we needed Slax in. That got made public.

This requires more permissions, which I don't think we really want. @Dacello do you agree
